### PR TITLE
feat: Complete Doctrine ORM state processor example

### PR DESCRIPTION
**Pull request by herndlm**
I was having problems with what was described there because it basically broke entity deletion, see https://github.com/api-platform/core/discussions/5107 for details.

Does this make sense? Not entirely sure if it counts as fix or feature, just let me know, I'll update branches accordingly if necessary.

It's a mix of concerns, but I also adapted the example to use property promotion because 3.0 is requiring PHP 8.1 anyways, right?

### CODE REVIEW

**Comment by alanpoulain** [comment]
LGTM. Could you change the title since we are not really decorating a service?

**Comment by herndlm** [comment]
adapted, I wasn't sure what you prefer here and just force-pushed because this PR is quite small anyways

File: core/state-processors.md [hunk 1]
```diff
@@ -94,25 +94,25 @@ Here is an implementation example which sends new users a welcome email after a
...
+    public function __construct(private ProcessorInterface $persistProcessor, private ProcessorInterface $removeProcessor, MailerInterface $mailer)
     {
-        $this->decorated = $decorated;
-        $this->mailer = $mailer;
     }
 
     public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $result = $this->decorated->process($data, $operation, $uriVariables, $context);
+        if ($operation instanceof Delete) {
```
Comment (soyuka) [inline core/state-processors.md:111]: DeleteOperationInterface

Comment (herndlm) [reply core/state-processors.md]: of course, thx. adapted

**Comment by alanpoulain** [comment]
Thanks!

**Pull request merged**